### ISSUE TITLE
About page-turning judgment

### ISSUE THREAD

**Issue by choristery** [oleksandrbalan/pagecurl#24]
Hi, I've encountered a habitual issue in usage. Currently, `dragForwardInteraction/dragBackwardInteraction`  is recognized through setting `Start` and `End` Rect for page flipping. However, in practical use, many users have provided feedback that they would prefer recognition of the previous or next page based on the direction of the gesture swipe, rather than being restricted within a fixed area. Is there any good way to set this up?

**Comment by oleksandrbalan** [comment]
Hi 👋 

Yea, it perfectly makes sense, but unfortunately it is not possible with the current API. Moreover current implementation relies on where tap has started to determine whether next & current or prev & current pages should be rendered. Maybe it could be refactored to wait for the gesture to happen and then take correct pages to render, I will try to look at it 👀

**Comment by oleksandrbalan** [comment]
@choristery I have managed to come up with some implementation, which as for me works pretty decent 👀 

Could you please try a behavior in the demo app on `feature/new-drag-gesture-interaction` branch? The new configuration could be set via `dragInteraction` param in `rememberPageCurlConfig`:

```kotlin
PageCurl(
    count = pages.size,
    config = rememberPageCurlConfig(
        dragInteraction = PageCurlConfig.GestureDragInteraction()
    )
) { index ->
    HowToPage(index, pages[index])
}
```

I opened a PR #25, but I would like to hear (or better say read 😅) your opinions before merging it to main branch.

Also I have to change API of `rememberPageCurlConfig` a little bit to support another drag interaction.

**Comment by choristery** [comment]
I think the new version of gesture interaction is very well-executed, aligning with user expectations during usage. However, there are areas for improvement in the UI interaction. Currently, the edges of the rolled-up pages do not seem to follow the finger's position on the screen. This gives users a sense of being out of control. When the finger is pressed and begins to move, the raised edge of the page should always follow the finger's position. This would provide a better sense of control.

**Comment by oleksandrbalan** [comment]
Thx for checking! 

However I am not sure that I fully understand what do you mean by "following the finger's position on the screen". Could you please record a video (possibly with visible touch positions)?

**Comment by choristery** [comment]
Hi, I uploaded a video on YouTube, hoping it can be helpful to you.

[Flip Style](https://youtube.com/shorts/8lnNZBrIl-s?si=sS4xvlPrM-RyvWR2)

**Comment by oleksandrbalan** [comment]
Ah, I see 🤔 

If I am not mistaken the PageCurl never behave like on that video. User's finger was always anchored to the page's flip (the edge between back-page of the current page and the content of next page).

I will think about how that behavior could be modified and will try to make it configurable. Could I ask you to open another issue to support such finger-on-page-edge behavior?

**Comment by oleksandrbalan** [comment]
Released in [v1.5.0](https://github.com/oleksandrbalan/pagecurl/releases/tag/v1.5.0) 🎉
